src/Compilers/Core/Portable/SpecialTypeExtensions.cs

```diff
@@ -57,6 +57,35 @@ public static bool IsBlittable(this SpecialType specialType)
             }
         }
 
+        /// <summary>
+        /// Checks if a type is an immutable value type primitive.
+        /// </summary>
+        public static bool IsImmutableValueTypePrimitive(this SpecialType specialType)
```

Comment:
That places an undocumented restriction on the future expansion of the `SpecialType` enumeration. While it's unlikely to ever be a problem, that's certainly a dangerous position to put yourself in when it can be avoided.